### DISCUSSION TITLE
running on GPU on a M1 mac

### DISCUSSION THREAD

**Discussion by ktyssowski** [neuropoly/axondeepseg#785] [Help needed]
I'm running ADS on a M1 mac and it works fine, but it's running on CPU. When it starts it says:
`2024-02-08 16:13:57.982 | INFO     | ivadomed.utils:define_device:160 - Cuda is not available.
2024-02-08 16:13:57.982 | INFO     | ivadomed.utils:define_device:161 - Working on cpu.`

Is it possible to use the GPU on an M1 mac? Should I just reinstall torch and make sure that is correct? Any ideas?

Thanks!

**Comment by mathieuboudreau** [comment]
Hi @ktyssowski - to my knowledge, only Nvidia GPUs are CUDA-compatible; is your mac running on one? You can check using these instructions: https://support.apple.com/en-ca/102312#:~:text=Check%20if%20the%20discrete%20or,GPUs%20your%20Mac%20might%20have.

**Reply by mathieuboudreau** [reply]
But besides that, the GPU really is only strongly recommended for *training* new models; if you're only looking to use existing models on your images to segment them, running the software on CPU will be fine, just slightly (not significantly) slower (unless you're planning on running this in a pipeline for thousands of images I'd reckon),

**Reply by ktyssowski** [reply]
Ok thanks! Yeah it doesn’t have nvidia—just wanted to check before I tried to troubleshoot!  Good to know cpu is not that much slower for running the models, too.
